examples/green_screen/main.cpp

```diff
@@ -0,0 +1,490 @@
+            break;
+        case 'f':
+            powerline_freq = std::stoi(optarg);
+            break;
+        case 'e':
+            color_exposure_usec = std::stoi(optarg);
+            break;
+        case 't':
+            depth_threshold = std::stoi(optarg);
+            break;
+        case '?':
+            throw std::runtime_error("Incorrect arguments");
+            break;
+        default:
+            throw std::runtime_error(std::string("getopt returned unexpected character code, ") +
+                                     std::to_string(input_opt));
+        }
+    }
+
+    if (chessboard_pattern.height == 0)
+    {
+        throw std::runtime_error("Chessboard height is not properly set! Use -h");
+    }
+    if (chessboard_pattern.width == 0)
+    {
+        throw std::runtime_error("Chessboard height is not properly set! Use -w");
+    }
+    if (chessboard_square_length == 0.)
+    {
+        throw std::runtime_error("Chessboard square size is not properly set! Use -s");
+    }
+
+    cout << "Chessboard height: " << chessboard_pattern.height << ". Chessboard width: " << chessboard_pattern.width
+         << ". Chessboard square length: " << chessboard_square_length << endl;
+
+    // Require 2 cameras
+    const int num_devices = k4a::device::get_installed_count();
+    if (num_devices != 2)
+    {
+        throw std::runtime_error("Exactly 2 cameras are required!");
+    }
+
+    // Set up a MultiDeviceCapturer to handle getting many synchronous captures
+    vector<int> device_indices{ 0, 1 };
+    MultiDeviceCapturer capturer(device_indices, color_exposure_usec, powerline_freq);
+
+    // Create configurations. Note that the main MUST be first, and the order of indices in device_indices is not
+    // necessarily preserved because the device_indices may not have ordered according to main and backup
+    vector<k4a_device_configuration_t> configs{ get_main_config(), get_backup_config() };
+    k4a_device_configuration_t &main_config = configs.front();
+    k4a_device_configuration_t &backup_config = configs.back();
+
+    capturer.start_devices(configs);
+
+    // This wraps all the device-to-device details
+    Transformation tr_color_backup_to_color_main =
+        calibrate_devices(capturer, configs, chessboard_pattern, chessboard_square_length);
+
+    k4a::calibration main_calibration = capturer.devices[0].get_calibration(main_config.depth_mode,
+                                                                            main_config.color_resolution);
+    k4a::calibration backup_calibration = capturer.devices[1].get_calibration(backup_config.depth_mode,
+                                                                              backup_config.color_resolution);
+    // Get the transformation from backup depth to backup color using its calibration object
+    Transformation tr_depth_backup_to_color_backup = get_depth_to_color_transformation_from_calibration(
```

Comment:
>tr_depth_backup_to_color_backup [](start = 19, length = 31)

suggestion: consolidate the naming convention of the <devicename>_<cameraname>, e.g. here you are using depth_backup and color_backup, somewhere below you are using main_depth and main_color, to help the user less confused, it might be good you are using the same order you like to use.

Reply:
Fixed in the latest commit